src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java

```diff
@@ -423,8 +423,10 @@ else if (withinAPod)
                             this.defaultedServiceAccountDirectory,
                             this.serviceAccountName,
                             this.defaultedServiceAccountName, this.clientId,
-                            this.defaultedClientId, this.clientSecret,
-                            this.defaultedClientSecret, this.redirectURL,
+                            this.defaultedClientId,
+                            this.clientSecret.substring(0, 5) + ".......",
+                            this.defaultedClientSecret(0, 5) + ".......",
+                            this.redirectURL,
```

Comment:
You got a compile error:

```
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /data/src/github.com/openshift/jenkins-openshift-login-plugin/src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java:[427,45] error: cannot find symbol
[ERROR]   symbol:   method substring(int,int)
  location: variable clientSecret of type Secret
/data/src/github.com/openshift/jenkins-openshift-login-plugin/src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java:[428,32] error: cannot find symbol
[INFO] 2 errors 

```

`clientSecret` is of type hudson.util.Secret

Use `Secret.toString(this.clientSecret).substring(0, 5) + "......."`

it accounts for null, but doesn't further encode (to again make diagnostics more difficult)

Reply:
Silly me, and me thinking it was a quick fix, one should never underestimate a quick fix when  not skilled in the say language :) i'll use a proper ide and send an update, thanks for your patience again.

Reply:
no problem and thanks for the effort